coldfront/core/project/utils_/removal_utils.py

```diff
@@ -132,3 +144,272 @@ def send_emails(self):
+        for user in unique_users_to_email:
+            template_context['user_first_name'] = user.first_name
+            template_context['user_last_name'] = user.last_name
+            try:
+                send_email_template(
+                    'Project Removal Request Completed',
+                    'email/project_removal/project_removal_complete.txt',
+                    template_context,
+                    email_sender,
+                    [user.email])
+            except Exception as e:
+                message = (
+                    f'Failed to send a notification email to {user.email}. '
+                    f'Details: \n{e}')
+                logger.exception(message)
+                num_failures += 1
+        return num_failures == 0
+
+    def _send_emails_safe(self):
+        """Send emails.
+
+        Catch all exceptions to prevent rolling back any
+        enclosing transaction.
+
+        If send failures occur, store a warning message.
+        """
+        try:
+            num_failures = self._send_emails()
+        except Exception as e:
+            message = (
+                f'Encountered unexpected exception when sending notification '
+                f'emails. Details: \n{e}')
+            logger.exception(message)
+        else:
+            if num_failures > 0:
+                message = f'Failed to send {num_failures} notification emails.'
+                self._warning_messages.append(message)
+
+
+class ProjectRemovalRequestUpdateRunner(object):
+    """An object that performs necessary database changes when a new
+    project request is approved."""
+
+    def __init__(self, request_obj):
+        self.request_obj = request_obj
+        self.removed_user = request_obj.project_user.user
+        self.project = request_obj.project_user.project
+        # A list of messages to display to the user.
+        self.error_messages = []
+        self.success_messages = []
+
+    def update_request(self, status):
+        project_removal_status_choice, _ = \
+            ProjectUserRemovalRequestStatusChoice.objects.get_or_create(
+                name=status)
+        self.request_obj.status = project_removal_status_choice
+        self.request_obj.save()
+
+    def complete_request(self, completion_time):
+        try:
+            self.request_obj.completion_time = completion_time
+            self.request_obj.save()
+
+            project_user_status_removed, _ = \
+                ProjectUserStatusChoice.objects.get_or_create(
+                    name='Removed')
+            self.request_obj.project_user.status = project_user_status_removed
+            self.request_obj.project_user.save()
+        except Exception as e:
+            message = f'Unexpected error setting fields for project removal ' \
+                      f'request {self.request_obj.pk}.'
+            self.error_messages.append(message)
+
+        try:
+            allocation_obj = Allocation.objects.get(project=self.project)
```

Comment:
@jofeinstein FYI: This would be problematic if the `Project` had a secure directory allocation.

I'm refactoring this class into `ProjectRemovalRequestProcessingRunner`, where I'm just removing the `User` from the compute `Allocation`.